src/tests/JIT/opt/InstructionCombining/NegMulOrDivToConst.cs

```diff
@@ -0,0 +1,337 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Test the following optimizations:
+//  -(v * const) => v * -const;
+//  -v * const => v * -const;
+//  -(v / const) => v / -const;
+//  -v / const => v / -const;
+//
+// Note that C# spec tells that `int.MinValue / -1` result is implementation specific, but
+// ecma-335 requires it to throw `System.ArithmeticException` in such case, so we should not
+// change 1 and -1 sign.
+
+using System;
+using System.Runtime.CompilerServices;
+
+namespace TestIntLimits
+{
+    class Program
+    {
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        static int CheckMulNeg()
+        {
+            bool fail = false;
+            if (Mul1_1(3) != -7 * 3)
```

Comment:
minor nit: can we make the method names `Mul`_1` and other more meaningful like `Mul_Minus7`, etc.?

Reply:
Thanks @sandreenko ! Should be good to go.